packages/core/src/utils/result.ts

```diff
@@ -146,7 +150,7 @@ export const mergeResultPatch = (
       }
     }
   } else {
-    withData.data = nextResult.data || prevResult.data;
+    withData.data = (nextResult.payload || nextResult).data || prevResult.data;
     errors = (nextResult.errors as any[]) || errors;
```

Comment:
needs to combine `result.errors` with `result.payload.errors` potentially

Reply:
`result.payload` doesn't have the `errors` property right?

Reply:
https://github.com/urql-graphql/urql/pull/3499/commits/fd6b15b10f1991bface15814387784443f93dc68